ansible/provision-and-execute-tests/playbook.yml

```diff
@@ -161,48 +181,65 @@
+      args:
+        chdir: '{{ test_suite_folder }}'
 
-    - name: NPM version result (to be removed)
-      debug:
-        var: npm_version_out
+    - name: Remove previous reports
+      ansible.windows.win_file:
+        path: '{{ test_suite_folder }}\reports'
+        state: absent
 
-    - name: Run Node version
-      ansible.windows.win_shell: "nvm use {{ node_version }} ; node -v"
-      register: node_version_out
+    - name: Run test-suite
+      ansible.windows.win_shell: "nvm use {{ node_version }} ; npm run test"
+      args:
+        chdir: '{{ test_suite_folder }}'
+        creates: '{{ test_suite_folder }}\reports'
+      environment:
+        LOGGING_ENDPOINT: https://staging-log-api.newrelic.com/log/v1
+        NERD_GRAPH_URL: https://staging-api.newrelic.com/graphql
+        API_KEY: "{{ new_relic_api_key }}"
+        ACCOUNT_ID: "{{ new_relic_account_id }}"
+        MONITORED_FILE: '{{ monitored_file }}'
+#        MONITORED_TCP_PORT: "{{ monitored_tcp_port }}" # waiting for https://github.com/fluent/fluent-bit/issues/6671
+        MONITORED_WINDOWS_LOG_NAME_USING_WINLOG: "{{ monitored_windows_log_name_using_winevtlog }}"
+        MONITORED_WINDOWS_LOG_NAME_USING_WINEVTLOG: "{{ monitored_windows_log_name_using_winlog }}"
 
-    - name: Node version result (to be removed)
-      debug:
-        var: node_version_out
+    - name: Fetch results
+      fetch:
+        flat: true # behave like copy, we only want the file, not the whole path
+        src: '{{ test_suite_report_path }}'
+        dest: '{{ test_reports_dir }}/{{ fb_version }}_{{ os_distro }}_{{ os_version }}_{{ arch }}.xml'
 
 - name: Process and merge test reports
   hosts: localhost
-  become: true
   vars:
     node_version: 18.18.0
+    controller_scripts_folder: ../../integration-tests/controller-scripts
     test_reports_dir: /tmp/test-reports
     combined_test_report_name: report.xml
   tasks:
-    - name: Set up node and trigger process
+    - name: Install node and script dependencies
       ansible.builtin.import_role:
         role: morgangraphics.ansible_role_nvm
       vars:
         nodejs_version: "{{ node_version }}"
-        # CentOS come without "wget", so using "curl" instead
+        # CentOS comes without "wget", so using "curl" instead
         nvm_install: "curl"
         nvm_commands:
-          - "cd ~/fluent-bit-package/integration-tests/controller-scripts \
-          && nvm exec default npm install \
-          && TEST_REPORT_ROOT_PATH={{ test_reports_dir }} \
-          TEST_REPORT_NAME={{ combined_test_report_name }} \ 
-          nvm exec default npm run"
-
-    - name: Capture merged test results
-      command: "cat {{ test_reports_dir }}/{{ combined_test_report_name }}"
-      register: report
-
-    - name: Display merged test results
-      debug:
-        msg: "{{ report.stdout }}"
+          - "cd {{ controller_scripts_folder }} && nvm exec default npm install"
```

Comment:
You can maybe use `chdir` instead of `cd ... &&`?

Reply:
Not available in `ansible.builtin.import_role`